common/persistence/visibility/store/sql/query_converter.go

```diff
@@ -179,7 +183,25 @@ func (c *QueryConverter) convertSelectStmt(sel *sqlparser.Select) error {
 	}
 
 	if sel.Where != nil {
```

Comment:
When might Where be nil? (I haven't really read any of this code yet so sorry if it's a dumb question)

Is this change enough to make things work for old-style `ListOpenWorkflowExecutionsByType`, etc. as well as string queries?

Reply:
If the user query string is empty.

Reply:
Hm, but we do want to add the constraint in that case. Basically in all cases unless they specifically query for a different division

Reply:
@dnr Fixed the query converter, and visibility store to handle namespace division field.

Reply:
I don't have enough context to review the whole thing but that part looks good.

Unfortunately we don't have a standalone integration test for that, since only ES supported it at all before, but ScheduleIntegrationSuite exercises the NamespaceDivision stuff (as Alex discovered)